php7/README

```diff
@@ -0,0 +1,22 @@
+================================================================================
+  PHP 7.3 yacc grammar
+================================================================================
+
+This is an adaption of PHP's 7.3 Yacc grammar for lrpar. There are few things
+that are currently not supported. First, the dangling else problem in PHP is
+resolved by adding the non-existent `T_NOELSE` token, which takes precedence
+over `else`. But since this token can't exist, shifting `else` is chosen in
+favour of a reduction. Since lrpar requires all tokens to be defined in the
+lexer, the `T_NOELSE` token has been removed. However, since lrpar by default
+resolves conflicts in favour of shifts, we get the same outcome, though will
+produce a warning.
+
+Second, PHP lexes strings differently, depending on their contents. For
+example, a string containing variables, like `"hello ${name}"`, will be parsed
+as `T_STRING_VARNAME`. In this grammar, currently all strings are lexed into
+`T_CONSTANT_ENCAPSED_STRING`. Also currently not supported are HEREDOCs and
```

Comment:
Is it bad that they're all T_CONSTANT_ENCAPSED_STRING? I'm not really sure what the consequence of this is.

Reply:
Maybe also rephrase the last sentence to "Also note that HEREDOCs and inlined HTML are not supported by the lexer at all."

Reply:
I'm not really sure what the consequence of this is either. There are some rules that only work with `T_STRING_VARNAME` and thus won't work if we parse something as `T_CONSTANT_ENCAPSTED_STRING`. Like for example `encaps_list`, which itself is referenced within a rule `scalar: '"' encaps_list '"'`. So PHP does some parsing within strings to decide what type it is. Something that is currently impossible in lrpar.

Reply:
Right, so that's a limitation of using lrlex's regular expression based lexer. So really what we want to say is something like "Some PHP strings can't be parsed with lrlex's purely regular expression based lexer. For example, ..."?